### ISSUE TITLE
Saving validation and training loss across training iterations

### ISSUE THREAD

**Issue by tiffanymtang** [ygong2501/calcium#1]
During training of deep neural networks, it can be helpful to compute evaluation metrics (e.g,. some loss function, IOU) on both the training and validation datasets every XX steps or every XX epochs. If you *save* these outputs, you can plot them over the number of training steps/iterations to see the trajectory of the training/validation losses.

The trajectories of these curves can serve as a quick diagnostic. Some things to keep in mind:

- If the validation loss is increasing (i.e., getting worse), then this is indicative that there is overfitting.
- If the training/validation loss is not yet flattening out, then additional iterations might be necessary for the model to converge to a local minimum.
- If the training loss is flat throughout the training and never decreases (not even at the beginning), then the model is not learning anything.

**Comment by ygong2501** [comment]
[test](https://github.com/ygong2501/calcium/tree/main/core)
